src/lambda/sns_to_glue/app.py

```diff
@@ -1,50 +1,37 @@
 import json
-import logging
-import os
-
 import boto3
-import synapseclient
-
-SSM_PARAMETER_NAME = os.getenv('SSM_PARAMETER_NAME')
-GLUE_WORKFLOW_NAME = os.getenv('GLUE_WORKFLOW_NAME')
-synapseclient.core.cache.CACHE_ROOT_DIR = '/tmp/.synapseCache'
-
-logger = logging.getLogger(__name__)
-logger.setLevel(logging.DEBUG)
-
-def lambda_handler(sns_event, context):
-    logger.info(sns_event)
-    try:
-      for record in sns_event['Records']:
-        payload = record['Sns']
-        synapse_id = payload['MessageAttributes']['SynapseId']['Value']
-        ssm_client = boto3.client('ssm')
-        glue_client = boto3.client('glue')
-        token = ssm_client.get_parameter(
-          Name=SSM_PARAMETER_NAME,
-          WithDecryption=True)
-        s3_loc = get_s3_loc(
-          synapse_id=synapse_id,
-          auth_token=token['Parameter']['Value'])
-        logger.debug(f'S3 location info: {s3_loc}')
-        logger.debug(f'Start workflow {GLUE_WORKFLOW_NAME}')
-        workflow_run = glue_client.start_workflow_run(Name=GLUE_WORKFLOW_NAME)
-        glue_client.put_workflow_run_properties(
-          Name=GLUE_WORKFLOW_NAME,
-          RunId=workflow_run['RunId'],
-          RunProperties={
-            'source_bucket': s3_loc['bucket'],
-            'source_key': s3_loc['key']})
-    except Exception as e:
-      logger.error(f'An error occurred while processing SNS message: {e}')
 
 
-def get_s3_loc(synapse_id, auth_token):
-    syn = synapseclient.Synapse()
-    syn.login(authToken=auth_token, silent=True)
-    f = syn.get(synapse_id, downloadFile=False)
-    logger.debug(f'synapse get response: {f}')
-    bucket = f['_file_handle']['bucketName']
-    key = f['_file_handle']['key']
-    s3_loc = {'bucket': bucket, 'key': key}
-    return s3_loc
+def lambda_handler(event, context):
```

Comment:
ahh a lambda, i'm not sure if you are aware but Sceptre supports direct deployments of lambdas to AWS with the [sceptre-sam-handler](https://github.com/Sceptre/sceptre-sam-handler).  You may want to check it out as it may simplify/streamline your CI.

Reply:
@zaro0508 Thanks for telling us about this handler -- we'll make a JIRA to look into it. It may not be an easy switch for us since we are not using IT's standard process for uploading files to S3, and at first glance this handler only has a subset of the sam functionality. Have you updated the lambda-template repository to use it?